test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -646,7 +639,7 @@ def test_cgroup_cpuset_and_memory(self):
         jid = self.server.submit(j)
         a = {'job_state': 'R'}
         self.server.expect(JOB, a, jid)
-        self.server.status(JOB, [ATTR_o, ATTR_e], jid)
+        self.server.status(JOB, ATTR_o, jid)
```

Comment:
Adding comments since this is the closest line to line 644-656.
I added following function and replaced line 644-656 with that. tried on my setup and it is working fine.

def read_job_output(self, job_file):
        """
        Open the job output file and read the content
        """
        tmp_out = ''
        self.logger.info("Job .o file: %s" % job_file)
        tmp_file = job_file.split(':')[1]
        i = 0
        while not os.path.isfile(tmp_file) and i < 10:
            i += 1
            time.sleep(1)
        if (i >= 10):
            self.assertFalse(True, "%s file do no exist even after 10s" %tmp_file)
        fd = self.logutils.open_log(tmp_file, sudo=True)
        self.assertTrue(fd, "Error while opening the file")
        tmp_out = fd.read().splitlines()
        return str(tmp_out)

Reply:
an if followed by assertFalse(True) can usually be replaced with putting the if conditional in an assert.  So in this case assertTrue(i >= 10, <msg>)

Reply:
good idea Bhroam.

Reply:
Took some effort, but it's done.